docs/proposals/all-in-one-volume.md

```diff
@@ -0,0 +1,258 @@
+the location within a volume for each resource is also available with the new
+single volume.
+
+## Current State Overview
+
+The only way of utilizing secrets, configmaps, and downward API currently is
+to access the data using separate pathing as shown in the volumeMounts section
+below:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: volume-test
+spec:
+  containers:
+  - name: container-test
+    image: jpeeler/scratch
+    volumeMounts:
+    - name: mysecret
+      mountPath: "/secrets"
+      readOnly: true
+    - name: podInfo
+      mountPath: "/podinfo"
+      readOnly: true
+    - name: config-volume
+      mountPath: "/config"
+      readOnly: true
+  volumes:
+  - name: mysecret
+    secret:
+      secretName: jpeeler-db-secret
+      items:
+      - key: username
+        path: my-group/my-username
+  - name: podInfo
+    downwardAPI:
+      items:
+        - path: "labels"
+          fieldRef:
+            fieldPath: metadata.labels
+        - path: "annotations"
+          fieldRef:
+            fieldPath: metadata.annotations
+  - name: config-volume
+    configMap:
+      name: special-config
+      items:
+      - key: special.how
+        path: path/to/special-key
+```
+
+## Analysis
+
+There are several combinations of resources that can be used at once, which 
+all warrant consideration. The combinations are listed with one instance of
+each resource, but real world usage will support multiple instances of a
+specific resource too.
+
+### ConfigMap + Secrets + Downward API
+
+The user wishes to deploy containers with configuration data that includes
+passwords. An application using these resources could be deploying OpenStack
+on Kubernetes. The configuration data may need to be assembled differently
+depending on if the services are going to be used for production or for
+testing. If a pod is labeled with production or testing, the downward API
+selector metadata.labels can be used to produce the correct OpenStack configs.
+
+### ConfigMap + Secrets
+
+Again, the user wishes to deploy containers with configuration data that
+includes passwords. In this case with MariaDB running, the operator may wish
+the container to have a ~/.my.cnf file that includes the username and password
+for the database.
+
+### ConfigMap + Downward API
+
+In this case, the user wishes to generate a config including the pod’s name
+(available via the metadata.name selector). This application may then pass the
+pod name along with requests in order to easily determine the source without
+using IP tracking.
+
+### Secrets + Downward API
+
+A user may wish to use a secret as a public key to encrypt the namespace of
+the pod (available via the metadata.namespace selector). This example may be
+the most contrived, but perhaps the operator wishes to use the application to
+deliver the namespace information securely without using an encrypted
+transport.
+
+### Resources configured with the same path
```

Comment:
I would not say this is 'configured' -- I would call this section 'Collisions between keys when all keys from a resource are projected'